eth/sync.go

```diff
@@ -184,22 +184,29 @@ func (cs *chainSyncer) handlePeerEvent(peer *eth.Peer) bool {
 	}
 }
 
-// loop runs in its own goroutine and launches the sync when necessary.
-func (cs *chainSyncer) loop() {
+// startFetcher owns the blockFetcher and txFetcher
+func (cs *chainSyncer) startFetcher() {
 	nodeCtx := common.NodeLocation.Context()
-	defer cs.handler.wg.Done()
-
 	cs.handler.blockFetcher.Start()
 	if nodeCtx == common.ZONE_CTX && cs.handler.core.ProcessingState() {
 		cs.handler.txFetcher.Start()
 		defer cs.handler.txFetcher.Stop()
 	}
+	defer cs.handler.wg.Done()
```

Comment:
why did this move? its best practice to have your defers as early as possible (usually the first line of code)

Reply:
moved